### PR TITLE
Ongage

### PR DESCRIPTION
**Pull request by sirpy**
# Description
- removed mautic
- switched to using ongage

About #351

### CODE REVIEW

File: src/server/crm/ongage.js [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+    this.baseHeaders = {
+      'Content-Type': 'application/json',
+      X_USERNAME: apikey,
+      X_PASSWORD: apisecret,
+      X_ACCOUNT_CODE: account_code
+    }
+  }
+
+  baseQuery(url, headers, body = null, method = 'post', timeout = 15000) {
+    const { baseUrl, log } = this
```
Comment (johnsmith-gooddollar) [inline src/server/crm/ongage.js:86]: if you will use axios, you won't need this method.
you could create axios intsance and configure all this stuff with request / response / error handlers
then just call .get / .post etc of `this.axios`

File: src/server/verification/verificationAPI.js [hunk 2]
```diff
@@ -13,8 +13,9 @@ import requestTimeout from '../utils/timeout'
 // import fuseapi from '../utils/fuseapi'
 import OTP from '../../imports/otp'
 import conf from '../server.config'
-import { Mautic } from '../mautic/mauticAPI'
+import { OnGageAPI } from '../crm/ongage'
```
Comment (johnsmith-gooddollar) [inline src/server/verification/verificationAPI.js:16]: maybe rename it to `Ongage`

**Review by johnsmith-gooddollar — CHANGES_REQUESTED** [review]

File: src/server/crm/ongage.js [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+  }, {})
+}
+
+export class OnGage implements CRMAPI {
+  baseUrl: string
+  log = logger.child({ from: 'OnGage' })
+
+  constructor(apiurl: string, account_code: string, apikey: string, apisecret: string) {
+    this.baseUrl = apiurl
+    this.baseHeaders = {
```
[the remark this answers is not in the document]
Comment (sirpy) [reply src/server/crm/ongage.js:77]: ok. can you implement all your suggested changes?

[the remark this answers is not in the document]
Comment (johnsmith-gooddollar) [reply src/server/crm/ongage.js:77]: ok

File: src/server/crm/ongage.js [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+
+const fieldsMap = {
+  firstName: 'first_name',
+  lastName: 'last_name',
+  fullName: 'fullname',
+  regMethod: 'regmethod',
+  torusProvider: 'torusprovider'
+}
+
+export const parseUtmString = utmString => {
```
[the remark this answers is not in the document]
Comment (johnsmith-gooddollar) [reply src/server/crm/ongage.js]: done

File: src/server/crm/ongage.js (hunk 1 above)
Comment (johnsmith-gooddollar) [reply src/server/crm/ongage.js]: done

File: src/server/verification/verificationAPI.js (hunk 2 above)
Comment (johnsmith-gooddollar) [reply src/server/verification/verificationAPI.js:16]: done

**Review by johnsmith-gooddollar — APPROVED** [review]

File: src/server/crm/ongage.js [hunk 5]
```diff
@@ -0,0 +1,228 @@
...
+    )
+  }
+
+  _configureClient(Config, log) {
+    const { ongageUrl, ongageAccount, ongageKey, ongageSecret } = Config
+
+    const httpClientOptions = {
+      validateStatus: status => status < 300,
+      baseURL: ongageUrl,
+      timeout: 15000,
```
Comment (sirpy) [inline src/server/crm/ongage.js:162]:
```suggestion
      timeout: 3000,
```

**Comment by sirpy** [comment]
@johnsmith-gooddollar tests no longer pass

**Comment by johnsmith-gooddollar** [comment]
> @johnsmith-gooddollar tests no longer pass

fixed

**Comment by johnsmith-gooddollar** [comment]
test were added, everything checked
ready to merge after new signup release

**Pull request merged**